### PR TITLE
Download most JVMs from Docker (except ORACLE8), and upgrade to latest version

### PR DESCRIPTION
**Pull request by luhenry**

### CODE REVIEW

**Review by bantonsson — APPROVED** [review]
Don't know if you know that you can test this in CI by changing image to `datadog/dd-trace-java-docker-build:luhenry_upgrade-java `

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,13 @@
 # Build from circleci image that uses current debian
+FROM debian:buster AS oracle8
```
Comment (bantonsson) [inline Dockerfile:2]: 👍🏻

**Review by devinsba — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -10,61 +10,31 @@ RUN set -eux; \
...
-    ZULU_REPO_DEB=https://cdn.azul.com/zulu/bin/zulu-repo_1.0.0-2_all.deb; \
-    SUM='b8d11979d9b1959b5ff621f1021ff0dba40c7d47d948ae6ec4a4bbde98cf71f5'; \
-    sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 0xb1998361219bd9c9; \
-    wget -q -O /tmp/zulu-repo.deb ${ZULU_REPO_DEB}; \
-    echo "${SUM} /tmp/zulu-repo.deb" | sha256sum -c -; \
-    sudo dpkg -i /tmp/zulu-repo.deb; \
-    sudo apt-get update; \
-    sudo apt-get install zulu7 zulu8 zulu11 zulu13 zulu15;
+COPY --from=openjdk:8-jdk-buster /usr/local/openjdk-8 /usr/lib/jvm/openjdk8
+COPY --from=openjdk:11-jdk-buster /usr/local/openjdk-11 /usr/lib/jvm/openjdk11
```
[the remark this answers is not in the document]
Comment (luhenry) [reply Dockerfile:23]: There won't be maven or gradle build tools coming from https://github.com/CircleCI-Public/cimg-openjdk/blob/main/11.0/Dockerfile#L30-L55. Wouldn't that be a problem?

[the remark this answers is not in the document]
Comment (bantonsson) [reply Dockerfile:23]: I think that we use the gradle wrapper and maven wrapper in all places, and they will download the relevant jars.

[the remark this answers is not in the document]
Comment (luhenry) [reply Dockerfile:23]: Ok, I'll update the base image then and see if anything breaks.

[the remark this answers is not in the document]
Comment (luhenry) [reply Dockerfile:24]: Some small fixes needed for the Maven wrapper, but that's it.

**Pull request merged**